src/components/Pill/PillButtonEnhanced/PillButtonEnhanced.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { IoIosArrowDown, IoMdClose } from 'react-icons/io';
+import { bem } from '../../../utils';
+import { ENTER_KEY } from '../../../constants';
+import { PillButtonBaseProps } from '../PillButton';
+import styles from './PillButtonEnhanced.scss';
+
+export interface Props extends PillButtonBaseProps {
+    /** Label/indicator in case of multiple selection, such as `+2` for instance */
+    multiSelectionLabel?: string;
+    /** Tooltip content to be shown when multiple selections are shown */
+    multiSelectionTooltip?: string;
+    /** Props needed for rendering optional priority selector */
+    prioritySelector?: boolean; // boolean is temporary placeholder until PrioritySelector component is finalized
+}
```

Comment:
- `isOpen, toggleDropdown, name, content`:  I kept the naming of all the "base" props from the other pill button, so that it is easy to migrate, and we don't introduce any breaking changes. so now all you have to do is add `variant: 'enhanced'` to an existing pill and it will work. This is the approach we agreed on during refinement. Even thought I agree that some of the names could be revised, I do feel that having less breaking changes is also good. An intermediate step could be to have these things named differently only for the enhanced button and do some mapping inside the Pill component, but not sure of the benefits for that. In general the situation is that we don't use the PillButton directly in the components, and probably want to keep it that way. 
- I like the `addionalContent` as a name better then the `multiSelection` I used. Thanks for the idea. Shall it be then `additionalContentLabel` and `additionalContentTooltip`? I tested and for new lines you can use `text 1\nLine 2` so no need to a ReactNode there.
- For the truncation tooltip, I don't think we need a prop. It will be handled by the `title` attribute and will always show `content` (the truncated section). Does that work for you? Note: this is also a reason that I think it is better not to have the content as an array, it gives more flexibility for the application to manage the logic of what goes where.
- Sorry for the width thing, I thought I pushed it. I would add a `maxWidth?: string` prop. If it is not defined we use the default, if it is, then we can overwrite it on the style level. I think that will give cover all the use cases. What do you think?

Reply:
@mukiienko I keep thinking about it, and I don't really like the idea of using a tooltip for the content itself. It is true that it is showing up faster that way, but there are 2 drawbacks:
1. it will show up also if the content is not truncated (unless we somehow add code in JS instead of just CSS which seems to me an awful lot of complication)
2. the content will be there twice not only visually but also for screen readers.
What say you?

Reply:
`contentTooltip` will be optional. All that you mentioned also applies to the `title` attribute, doesn’t it?
1) the `title` is shown always, event the text are not the truncated, right?
2) we need duplicate text for `title` as well. For readers we can use appropriate `aria` attribute.
If you think more elegant way to implement via `title` that I don't mind. So, it will be `content={<span title="some text">Some text</span>}`, right?
@eszthoff

Reply:
@mukiienko 
1. yes, but since it has a delay, it is not that evident. You need to want to see it for it show up.
2. I think screen readers are better equipped to understand `title` in this scenario, but I haven't tested it indeed.